### PR TITLE
Bug 1420267 - New (about:blank) tab instead of Google Maps app

### PR DESCRIPTION
**Pull request by justindarc**

### CODE REVIEW

**Comment by farhanpatel** [comment]
@justindarc you mentioned `the website is trying to open a link to Google Maps in a new tab without passing along any referrer info` Can we not set it in `createWebViewWith`. The `WKNavigationAction` contains info about the request opening the new window.

**Comment by farhanpatel** [comment]
Also is this similar to https://github.com/mozilla-mobile/firefox-ios/issues/2481

File: Client/Frontend/Browser/WindowHelper.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    class func name() -> String {
+        return "WindowHelper"
+    }
+
+    required init(tab: Tab) {
+        self.tab = tab
+        if let path = Bundle.main.path(forResource: "WindowHelper", ofType: "js"), let source = try? NSString(contentsOfFile: path, encoding: String.Encoding.utf8.rawValue) as String {
+            let userScript = WKUserScript(source: source, injectionTime: WKUserScriptInjectionTime.atDocumentEnd, forMainFrameOnly: false)
+            tab.webView!.configuration.userContentController.addUserScript(userScript)
+        }
```
Comment (jhugman) [inline Client/Frontend/Browser/WindowHelper.swift]: Given that this is just injecting javascript, can we use `tab.injectUserScriptWith`, and do away with this class altogether?

File: Client/Assets/WindowHelper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    // ```
+                    // someLink.addEventListener('click', function(evt) {
+                    //     var newTab = window.open();
+                    //     newTab.opener = null;
+                    //     newTab.location = someLink.href;
+                    //     return false;
+                    // });
+                    // ```
+                    //
+                    // For more info, see Bug 1420267: https://bugzilla.mozilla.org/show_bug.cgi?id=1420267
```
Comment (jhugman) [inline Client/Assets/WindowHelper.js]: TODO: minify all these content scripts.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -1886,8 +1889,8 @@ extension BrowserViewController: TabManagerDelegate {
     }
 }
 
-/// List of schemes that are allowed to open a popup window
-private let SchemesAllowedToOpenPopups = ["http", "https", "javascript", "data"]
+/// List of schemes that are allowed to open as a popup window
+private let SchemesAllowedToOpenAsPopups = ["http", "https", "javascript", "data", "about"]
```
Comment (jhugman) [inline Client/Frontend/Browser/BrowserViewController.swift]: This is a significant change of meaning. Is this definitely the case?

**Pull request closed without merging**